test/e2e/framework/service/jig.go

```diff
@@ -558,20 +632,14 @@ func (j *TestJig) WaitForNewIngressIP(ctx context.Context, existingIP string, ti
 func (j *TestJig) ChangeServiceNodePort(ctx context.Context, initial int) (*v1.Service, error) {
 	var err error
 	var service *v1.Service
-	for i := 1; i < NodePortRange.Size; i++ {
-		offs1 := initial - NodePortRange.Base
-		offs2 := (offs1 + i) % NodePortRange.Size
-		newPort := NodePortRange.Base + offs2
-		service, err = j.UpdateService(ctx, func(s *v1.Service) {
-			s.Spec.Ports[0].NodePort = int32(newPort)
-		})
-		if err != nil && strings.Contains(err.Error(), errAllocated.Error()) {
-			framework.Logf("tried nodePort %d, but it is in use, will try another", newPort)
-			continue
-		}
-		// Otherwise err was nil or err was a real error
-		break
+	newport, ok := j.GetUnusedStaticNodePortAndReserve()
+	if !ok {
+		err = errors.New("could not find any unused nodeport")
+		return service, err
 	}
+	service, err = j.UpdateService(ctx, func(s *v1.Service) {
```

Comment:
we may try to retry a few times just in case the system has reserved that port, if the user has created a nodeport service manually per example.

Retry until `j.GetUnusedStaticNodePortAndReserve()` returns !ok

Reply:
This change has been reverted based on discussion [here](https://github.com/kubernetes/kubernetes/pull/127153#discussion_r1759585878) . Please let me know if this should not be the case, else I will resolve the conversation.